### PR TITLE
Functional Settings Page Edits

### PR DESCRIPTION
**Pull request by spiddubnyy**
Screenshot 2024-04-18 at 2 44 28 AM

### CODE REVIEW

File: UserService/app/routes.py [hunk 1]
```diff
@@ -93,6 +93,114 @@ def register_user():
...
+    #Perform validation
+    #check if user or email already in use
+    username = request.args.get('username')
+    current_user = User.query.filter_by(id=username).first()
+    if current_user == None:
+        abort(400, description="User does not exist")
+
+    company = Company.query.filter_by(id=current_user.company_id).first()
+
+    user_info = {
```
Comment (StuartGavidia) [inline UserService/app/routes.py:110]: Great retrieval of data!

File: UserService/app/routes.py [hunk 2]
```diff
@@ -93,6 +93,114 @@ def register_user():
...
+@routes.route("/users/updatePassword", methods=['POST'])
+def update_password():
+    """
+    This route registers a user
+    """
+    data = request.json  # Get JSON payload from the client
+    user_name = data.get('username', '')
+    password = data.get('password', '')
+    #hash password to prepare for storage
+    hashed_password = bcrypt.generate_password_hash(password).decode('utf-8')
```
Comment (StuartGavidia) [inline UserService/app/routes.py:188]: This is good, glad you caught the hashed password library.

File: UserService/app/routes.py [hunk 3]
```diff
@@ -93,6 +93,114 @@ def register_user():
...
+    record_to_update = User.query.filter_by(username=user_name).first()
+    record_to_update.password_hash = hashed_password
+
+    db.session.add(record_to_update)
+    db.session.commit()
+
+    #after succesfully registering user, clear the users from company cache
```
Comment (StuartGavidia) [inline UserService/app/routes.py:195]: Remove the commented code.

**Comment by StuartGavidia** [comment]
PR looks great, I left some comments, let me know if you have questions.

File: UserService/app/routes.py (hunk 3 above)
Comment (spiddubnyy) [reply UserService/app/routes.py]: removed.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
```
Comment (StuartGavidia) [inline package.json:2]: Looks like the package.json content got removed, you can paste one from a previous version and then ensure that the app still works

Comment (spiddubnyy) [reply package.json]: resolved

**Pull request merged**